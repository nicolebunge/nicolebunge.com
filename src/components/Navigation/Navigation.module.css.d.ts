declare const styles: {
  readonly navigation: string;
  readonly navigation__list: string;
  readonly 'navigation__list--inline': string;
  readonly navigation__link: string;
  readonly 'navigation__link--active': string;
};
export = styles;
