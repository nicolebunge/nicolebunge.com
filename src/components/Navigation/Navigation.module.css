.navigation {
  font-weight: 600;
}

.navigation__list {
  list-style: none;
  margin: 0;
  padding: 0;
}

.navigation__list--inline {
  display: flex;
  flex-wrap: wrap;
  gap: 1em;
}

.navigation__link {
  display: inline-block;
  padding-bottom: var(--space-md);
  padding-top: var(--space-md);
  position: relative;
  transition: color 0.3s;
}

.navigation__link--active {
  color: var(--color-primary);
}
