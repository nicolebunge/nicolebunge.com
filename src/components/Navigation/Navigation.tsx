/* eslint-disable camelcase */

import clsx from 'clsx';
import React, { HTMLAttributes } from 'react';
import Link from '../Link';
import {
  navigation,
  navigation__link,
  navigation__linkActive,
  navigation__list,
  navigation__listInline,
} from './Navigation.module.css';

interface Route {
  name: string;
  path: string;
}

export interface NavigationProps extends HTMLAttributes<HTMLDivElement> {
  inline?: boolean;
  routes: Route[];
}

function Navigation(props: NavigationProps): JSX.Element {
  const { className, inline, routes, ...otherProps } = props;

  return (
    <nav className={clsx(className, navigation)} {...otherProps}>
      {routes && (
        <ul
          className={clsx(navigation__list, {
            [navigation__listInline]: inline,
          })}
        >
          {routes.map((route) => (
            <li key={route.path}>
              <Link
                to={route.path}
                className={navigation__link}
                activeClassName={navigation__linkActive}
              >
                {route.name}
              </Link>
            </li>
          ))}
        </ul>
      )}
    </nav>
  );
}

export default Navigation;
