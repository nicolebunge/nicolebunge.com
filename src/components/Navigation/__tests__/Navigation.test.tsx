import { render } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';
import Navigation from '..';

describe('Navigation component', () => {
  it('renders correctly', async () => {
    const routes = [
      {
        name: 'Index',
        path: '/',
      },
    ];
    const { container } = render(<Navigation routes={routes} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
