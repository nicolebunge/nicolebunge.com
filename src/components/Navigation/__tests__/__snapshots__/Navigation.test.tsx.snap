// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders correctly 1`] = `
<nav
  class="navigation"
>
  <ul
    class="navigation__list"
  >
    <li
      class="navigation__item"
    >
      <a
        class="navigation__link"
        href="/"
      >
        Index
      </a>
    </li>
  </ul>
</nav>
`;
